pkg/adapter/adapter.go

```diff
@@ -302,30 +304,52 @@ func (a *Adapter) PollForMessages(channel *wabbit.Channel,
 	}
 }
 
+func setMsgContentType(msg wabbit.Delivery) string {
+	contentType := fmt.Sprint(msg.Headers()["Content-type"])
```

Comment:
its ok to use this or should i search for the datacontenttype header as well?

Reply:
I do not think it should matter for here since we're not really consuming the event. It should only be for the consumer that can decide how to decode the actual data.
https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#datacontenttype

Reply:
FYI, I had been wanting to use this protocol adapter here:
https://github.com/cloudevents/sdk-go/tree/82f2b61ecde41fd0577969cc58a7c2a18eeda250/protocol/amqp/v2

But it's for different version than what the rabbitmq supports. However, there's probably some interesting learnings there as well.

Reply:
FWIW, here's the spec for amqp 1.0 messaging formats for the CloudEvents.
https://github.com/cloudevents/spec/blob/v1.0.1/amqp-protocol-binding.md